### PR TITLE
MIGHTY MAN BRANDI BRUTAL HERE TO TEACH YOU ABOUT THE REAL INTER-GALACTIC HEAVYWEIGHT CHAMPION, YOURSELF: Adds a new version of Wrestling to the Championship belt.

### PR DESCRIPTION
**Pull request by Iamgoofball**
goon port #2: wrasslin belt

https://www.dropbox.com/s/8bctpxkjsxznr74/SNAP_INTO_A_SLIM_JIM.mp4?dl=0
wggjskc

:cl: "Macho Man" Randy Savage
rscadd: OOOHHH YEAAAAHHHH
rscadd: SNAP INTO A JLIM SIM WITH THE ALL NEW WRESTLING BELT
rscadd: YOU CAN FREAKOUT YOUR OPPONENTS WITH THE 5 HOT MOVES ON THIS BELT YEAH
rscadd: PRAY TO THE LOCAL SPACE GODS TO EXCHANGE YOUR TELECRYSTALS TO ENSURE THE CREAM RISES TO THE TOP
/:cl:

### CODE REVIEW

**Comment by coiax** [comment]
So uh, attempting to cut through the memes here:

A 7TC item that looks the same as the wrestling belt, but gives you the SLIM JIM(?) martial art, which gives you a bunch of moves, and should make you superior at hand to hand combat.

**Comment by Isratosh** [comment]
i love this :+1:

**Comment by Bawhoppen** [comment]
NO MORE TRAITOR ITEMS :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1:

**Comment by RemieRichards** [comment]
:-1: I prefer our wrestling martial art, lets me suplex the train.

**Comment by optimumtact** [comment]
>new goof pr
    >goof porting more goonshit

:neutral_face:

File: code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+	A.visible_message("<span class = 'danger'><B>[A] lifts [D] up!</B></span>")
+
+	spawn (0)
+		if (D)
+			animate(D, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
+		sleep (15)
+		if (D)
+			animate(transform = null, time = 1, loop = 0)
+
+	for (var/i = 0, i < 3, i++)
```
Comment (optimumtact) [inline code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm]: can you generalise this what I assume is rotation/spinning shit into some helper procs please

this wall of nested code makes me want to gouge out my eyes

**Comment by Aloraydrel** [comment]
yes please

**Comment by ToGWtF** [comment]
speedmerge

**Comment by Supermichael777** [comment]
Action button martial art is best martial art, fuck combos

**Comment by Aloraydrel** [comment]
Now just needs the soundfiles

**Comment by AdamElTablawy** [comment]
but isn't martial arts more the sleeping carp's thing along with melee immunity

File: tgstation.dme [hunk 2]
```diff
@@ -221,6 +221,7 @@
 #include "code\datums\helper_datums\teleport.dm"
 #include "code\datums\helper_datums\topic_input.dm"
 #include "code\datums\martial\krav_maga.dm"
+#include "code\datums\martial\SNAP_INTO_A_SLIM_JIM.dm"
```
Comment (Cheridan) [inline tgstation.dme]: slim jims are copyrighted

**Comment by PKPenguin321** [comment]
Needs the cooldowns from goon too, especially on the move that lets you gib people

File: tgstation.dme (hunk 2 above)
Comment (Iamgoofball) [inline tgstation.dme]: Whoops

**Comment by Iamgoofball** [comment]
@remierichards I'll just re-add suplexing as a move. Aside from suplexing our  current wrestling is shit, though.

File: code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+/datum/action/slam/Trigger()
+	if(owner.incapacitated())
+		owner << "<span class='warning'>You can't WRESTLE while you're OUT FOR THE COUNT.</span>"
+		return
+	owner.visible_message("<span class='danger'>[owner] prepares to BODY SLAM!</span>", "<b><i>Your next attack will be a BODY SLAM.</i></b>")
+	var/mob/living/carbon/human/H = owner
+	H.martial_art.streak = "slam"
+
+/datum/action/throw_wrassle
+	name = "Throw (Cinch) - Spin a cinched opponent around and throw them.."
```
Comment (PKPenguin321) [inline code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm]: Extra .

File: tgstation.dme (hunk 2 above)
Comment (Supermichael777) [inline tgstation.dme]: Their trademark (the bit that matters) does not extend into entertainment.
you would end up in more trouble from the macho man reference.

Comment (MrPerson) [inline tgstation.dme]: Either way rename the file to something more descriptive about what you're going to find. Not everyone knows every meme ever

File: code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+	var/turf/T = get_edge_target_turf(A, get_dir(A, get_step_away(D, A)))
+	if (T && isturf(T))
+		D.throw_at(T, 3, 2)
+		D.Weaken(1)
+
+/datum/martial_art/wrestling/proc/drop(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!D)
+		return
+	if(!D.lying)
+		A << "[D] needs to be prone to do this!"
```
Comment (MrPerson) [inline code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm]: return?

**Comment by MrPerson** [comment]
I don't like the joke so obviously I don't like this.

no_fun_allowed.jpg etc

**Comment by Niknakflak** [comment]
:-1:

**Comment by Durkel** [comment]
This is literally the best feature goon has, merge asap.

**Comment by PKPenguin321** [comment]
^
I haven't played goon much, but I have used the wrestling belt, and rampages with it are *very* satisfying.

Two things, though: Goon has less total TCs, so this should cost more than 7 on our server. Goon also has cooldowns on the moves, while this doesn't.

**Comment by Ultimate-Chimera** [comment]
:-1:

**Comment by Iamgoofball** [comment]
@Ultimate-Chimera explain, furry

**Comment by Zilenan91** [comment]
Yeah this should cost like 15 TCs at the least because of the gibbings

**Comment by Iamgoofball** [comment]
where does it gib

**Comment by Zilenan91** [comment]
Oh I thought this was literally the maetcho maenn wrestling moves rather than just the wrestling belt

**Comment by Niknakflak** [comment]
:-1:

**Comment by Robustin** [comment]
Why the fuck was this testmerged. This was the last straw for some people, getting murdered by some turbohitler lying flat on his back flipping around one-shotting people until the entire station was dead, I would've quit too if I were playing that round. 

Why the fuck isn't this stuff tested privately before we just impose it on all the server.

**Comment by Zilenan91** [comment]
Goof always makes his shit insane when he merges it then nerfs it afterwards

**Comment by Niknakflak** [comment]
:-1:

**Comment by Robustin** [comment]
I lost the quote on IRC but basically it was:

Goof: "Why should I care that it ruined one round, we have infinity more"

**Comment by PKPenguin321** [comment]
he's not wrong

**Comment by Robustin** [comment]
If a player leaves forever because they don't want to put up up with that kind of shit, we don't have infinity more to replace them. April Fools was a demonstration that excessive shittery can drive away players.

**Comment by Iamgoofball** [comment]
they aren't going to leave forever

they might get off for the day but they'll be back tomorrow

**Comment by Iamgoofball** [comment]
addendum: if someone was going to leave the server completely they're probably approaching their ss13 burnout anyways and will probably quit playing in genneral

**Comment by MrPerson** [comment]
That attitude is bullshit goofball.

**Comment by AdamElTablawy** [comment]
if you're a salty enough asshole that someone backflipping around the station kicking your shit in yelling about snapping into slim jims before suplexing the HoS is somehow game ruining for you and enough to make you leave the server you're quite frankly the most dull asshole in existence

**Comment by Zilenan91** [comment]
If it happens multiple rounds in a row by multiple people it stops being funny and starts being frustrating

**Comment by optimumtact** [comment]
The whole point of a test merge is to get feedback so it looked like it worked

**Comment by AdamElTablawy** [comment]
the feedback it got was 

'this sucks''

the proper response is 'okay how do we fix it'

if you expect a test merge to only be met with 'XDDDDDDD THIS IS THE BEST THING EVER' and only that that's quite frankly stupid because the reason testmerges exist is to collect positive or negative feedback and go from there


so what were the issues cited as to why people disliked it

**Comment by Iamgoofball** [comment]
So far its the suplex permastun and the icon issue. I'll patch those.

**Comment by optimumtact** [comment]
I didn't expect anything

**Comment by Robustin** [comment]
There's a lot more, my beef was "Suplex permastun, low TC, and icon issue" was something I could've deduced in about 30 seconds if it was merged as an admin-only item. Also you can knee-drop people who aren't even prone.

The subtler feedback that can only come from full rounds, that's what test-merging is for, not "I need the howling chorus of dozens of pissed of players before I realize that broken icons + murderboning is a bad thing".

**Comment by AdamElTablawy** [comment]
TC seems to be a major point.

Tbh I would make it cost as much as the sleeping carp scroll-Or perhaps merge the two into a beacon of some sort that lets you warp in either the carp scroll or the belt.

**Comment by Iamgoofball** [comment]
Suplex has been removed.
TC count upped to 14.
Icon bug fixed.

**Comment by Niknakflak** [comment]
:-1:

File: tgstation.dme (hunk 2 above)
Comment (Iamgoofball) [inline tgstation.dme]: well duh

**Comment by Dorsisdwarf** [comment]
It's actually pretty funny, I quite like it.

File: code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm]: No, this is pretty wrasslin specific and stays here.

Comment (RemieRichards) [inline code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm]: ```for(var/i in 1 to 3)```

**Comment by KorPhaeron** [comment]
@Iamgoofball 

Do you want to compromise and make it an admin only thing and see how it plays out over a couple weeks via traitor TC trades before revisiting it being in the uplink? It's how I snuck stands in

**Comment by Iamgoofball** [comment]
@KorPhaeron We've been testmerging them as traitor items over the last few weeks and no one's come complaining, so I think it's fine to merge as a traitor item for now.

inb4 everyone crawls out of the woodworks to complain now

**Comment by KorPhaeron** [comment]
The head coder gave you some reddit disapproval though

**Comment by Iamgoofball** [comment]
he hasn't elaborated on why so I'm assuming it's meme

plus it's a bad thing if ratings actually effect PRs

**Comment by Robustin** [comment]
Just do the admin-only thing. You can get some actual feedback besides "lol we testmerged it for a few rounds and sometimes it didn't ruin them".

**Comment by KorPhaeron** [comment]
Fuckin goof is gonna let this die rather than be admin only isnt he

**Comment by Dorsisdwarf** [comment]
Then it'll die again. PR merge is not a right, unlike what like 90% of coders seem to think

**Comment by PKPenguin321** [comment]
If this dies I could salvage again

**Comment by Iamgoofball** [comment]
i don't understand why we need to make it admin only for testing

WHEN WE'VE ALREADY BEEN TESTING FOR WEEKS AND NO ONE IS GIVING ME FEEDBACK ASIDE FROM "ITS BAD K"

WHY WOULD MAKING IT ADMIN ONLY IMPROVE THE FEEDBACK AT ALL

**Comment by KorPhaeron** [comment]
>testing for weeks
>merged for like a day maybe

@PKPenguin321 

Sure go nuts

**Comment by Iamgoofball** [comment]
@KorPhaeron I've been working with Shadowlight.

Trust me, we've had it test merged multiple times. Why do you refuse to listen?

**Comment by Zilenan91** [comment]
After suplex got removed it was pretty decent, I could get behind having this in the game.

**Comment by PKPenguin321** [comment]
badaboom now its admin only

**Comment by duncathan** [comment]
how does wrestling outperform a revolver

**Comment by KorPhaeron** [comment]
@duncathan is this code okay?

**Comment by Ultimate-Chimera** [comment]
Goof PR, how is this still up? Lol

**Comment by Iamgoofball** [comment]
You never answered my question.

@KorPhaeron i made it admin only merge it already

**Comment by KorPhaeron** [comment]
I was waiting for someone to review the code

**Comment by Razharas** [comment]
Im cleaning pr list so no more "lets just ignore it" memes

**Comment by Iamgoofball** [comment]
# THIS HAS BEEN READY TO MERGE FOR WEEKS WHAT THE FUCK

**Comment by Robustin** [comment]
lol Goof, spend another 2 weeks in page 3 and you'll know what its like to be me

**Comment by Iamgoofball** [comment]
I'm gonna cheat soon

File: code/datums/martial/wrestling.dm [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+	if(owner.incapacitated())
+		owner << "<span class='warning'>You can't WRESTLE while you're OUT FOR THE COUNT.</span>"
+		return
+	owner.visible_message("<span class='danger'>[owner] prepares to LEG DROP!</span>", "<b><i>Your next attack will be a LEG DROP.</i></b>")
+	var/mob/living/carbon/human/H = owner
+	H.martial_art.streak = "drop"
+
+/datum/martial_art/wrestling/teach(var/mob/living/carbon/human/H,var/make_temporary=0)
+	..()
+	H << "<span class = 'userdanger'>SNAP INTO A JLIM SIM!</span>"
```
Comment (Cheridan) [inline code/datums/martial/wrestling.dm]: still dumb

**Pull request merged**